### PR TITLE
RestartPdsAgentPodAndPerformBackupAndRestore - Restart pds agent pod, ScaleUp Data service and Perform Backup and Restore

### PR DESCRIPTION
**Pull request by Madanagopal19**
Testcase - RestartPdsAgentPodAndPerformBackupAndRestore

**Which issue(s) this PR fixes** (optional)
Closes #DS-7993

Jenkins logs - https://jenkins.pwx.dev.purestorage.com/job/PDS/job/pds2.0-byoc-branch-build-test/866/consoleFull

### CODE REVIEW

File: config [hunk 1]
```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
```
Comment (atrivedi-px) [inline config:1]: Please remove this file

File: tests/unifiedPlatform/pds2/pds_restore_negative_test.go [hunk 2]
```diff
@@ -15,6 +15,141 @@ import (
...
+				log.InfoD(steplog)
+				backupResponse, err := WorkflowPDSBackup.GetLatestBackup(*deployment.Create.Meta.Uid)
+				log.FailOnError(err, "Error occured while creating backup")
+				backUpBeforeAgentRestart = *backupResponse.Meta.Uid
+				log.Infof("Latest backup ID [%s], Name [%s]", *backupResponse.Meta.Uid, *backupResponse.Meta.Name)
+				err = WorkflowPDSBackup.WaitForBackupToComplete(*backupResponse.Meta.Uid)
+				log.FailOnError(err, "Error occured while waiting for backup to complete")
+			})
+
+			steplog = "Restart PDS Agent Pods and Validate if it comes up"
```
Comment (atrivedi-px) [inline tests/unifiedPlatform/pds2/pds_restore_negative_test.go:66]: Should this be running in parallel with ScaleUp step?

File: config (hunk 1 above)
Comment (Madanagopal19) [reply config]: done

File: tests/unifiedPlatform/pds2/pds_restore_negative_test.go (hunk 2 above)
Comment (Madanagopal19) [reply tests/unifiedPlatform/pds2/pds_restore_negative_test.go:66]: This testcase is for sequential execution

**Review by atrivedi-px — APPROVED** [review]
LGTM

**Pull request merged**